drivers/bluetooth/hci/Kconfig

```diff
@@ -158,8 +158,7 @@ config BT_SILABS_EFR32
 	depends on ZEPHYR_HAL_SILABS_MODULE_BLOBS || BUILD_ONLY_NO_BLOBS
 	depends on !PM || SOC_GECKO_PM_BACKEND_PMGR
 	select SOC_GECKO_USE_RAIL
-	select MBEDTLS
-	select MBEDTLS_PSA_CRYPTO_C
+	select PSA_CRYPTO_PROVIDER
 	select MBEDTLS_ENTROPY_C
```

Comment:
Similar argument, but opposite conclusion: `MBEDTLS_ENTROPY_C` is actually not required for the Silicon Labs HCI driver -- it just needs `psa_generate_random()`, so this should be removed rather than re-adding an `MBEDTLS` dependency.